src/Orleans/AzureUtils/OrleansSiloInstanceManager.cs

```diff
@@ -168,15 +168,18 @@ private OrleansSiloInstanceManager(string deploymentId, string storageConnection
             {
                 await instance.storage.InitTableAsync()
                     .WithTimeout(initTimeout);
-
             }
-            catch (TimeoutException)
+            catch (TimeoutException te)
             {
-                instance.logger.Fail(ErrorCode.AzureTable_32, String.Format("Unable to create or connect to the Azure table in {0}", initTimeout));
+                string errorMsg = String.Format("Unable to create or connect to the Azure table in {0}", initTimeout);
+                instance.logger.Error(ErrorCode.AzureTable_32, errorMsg, te);
+                throw new OrleansException(errorMsg, te);
```

Comment:
That's a good question @sergeybykov 
Most of the problem here is that Logger/Fail is a particularly nasty way to fail ! 
It does cause Error to be written into Windows Event log, which might be best that can be done for "headless" running, but is definitely sub-optimal for tests cases - especially for test suites running on cloud build servers.

I suspect the optimal way this should work is that the exception gets passed back to the caller which is ultimately the hosting class [OrleansHost.exe or TestingSiloHost or OrleansAzure Silo or similar] which then should make decision how to handle the failure.

From silo startup perspective, I don't think there is not really any difference between this specific failure to connect to membership table, and any other exception thrown during silo initialization.

I'll take another look at the code paths. I am pretty sure that OrleanAzureSilo and TestingSiloHost already handle silo init exceptions correctly, but I'll need to double check for other *Host classes too.